libraries/redcore/api/soap/operation/operation.php

```diff
@@ -285,6 +285,7 @@ private function task($taskName, $data)
 
 		// We are setting the operation of the webservice to task
 		$this->webservice->options->set('task', $taskName);
+		JFactory::getApplication()->input->set('task', $taskName);
```

Comment:
for soap I think it is necessary... because in most frameworks you import wsdl with all procedure calls (rpcs) and if we would need to change path (path in the URI &task=translate) for every task operation then they would need to import wsdl with different route... which would just create unnecessary code and complexity...